### PR TITLE
SAMZA-860, SAMZA-984: Upgraded RocksDB version to 5.0.1 and added configuration for managing RocksDB logging.

### PR DESCRIPTION
**Pull request by prateekm**

### CODE REVIEW

File: samza-kv-rocksdb/src/main/scala/org/apache/samza/storage/kv/RocksDbKeyValueStore.scala [hunk 1]
```diff
@@ -139,7 +139,7 @@ class RocksDbKeyValueStore(
     metrics.puts.inc
     require(key != null, "Null key not allowed.")
     if (value == null) {
-      db.remove(writeOptions, key)
```
Comment (vjagadish) [inline samza-kv-rocksdb/src/main/scala/org/apache/samza/storage/kv/RocksDbKeyValueStore.scala]: Is `remove` API being deprecated over `delete` in `RocksDB`?
Similar question for `dispose` versus `close` APIs.

File: gradle/dependency-versions.gradle [hunk 2]
```diff
@@ -30,7 +30,7 @@
   metricsVersion = "2.2.0"
   kafkaVersion = "0.10.0.1"
   commonsHttpClientVersion = "3.1"
-  rocksdbVersion = "3.13.1"
```
Comment (vjagadish) [inline gradle/dependency-versions.gradle]: SAMZA-964 (continuous checkpointing for stores) seems to be unrelated to this upgrade. Is it a typo?

File: docs/learn/documentation/versioned/jobs/configuration-table.html [hunk 3]
```diff
@@ -1422,6 +1422,21 @@
...
+                <tr>
+                    <td class="property" id="stores-rocksdb-log-file-size">stores.<span class="store">store-name</span>.<br>rocksdb.max.log.file.size.bytes</td>
+                    <td class="default">67108864</td>
+                    <td class="description">
+                        The maximum size in bytes of the RocksDB LOG file before it is rotated.
+                    </td>
+                </tr>
+
+                <tr>
+                    <td class="property" id="stores-rocksdb-num-log-files">stores.<span class="store">store-name</span>.<br>rocksdb.keep.log.file.num</td>
```
Comment (vjagadish) [inline docs/learn/documentation/versioned/jobs/configuration-table.html]: minor: What do you think about renaming this property to `num.log.files`? A nice thing is that it's arguably consistent with the other Samza property `num.write.buffers`?  

(Or, is the goal to mirror rocksDb config keys?)

File: samza-kv-rocksdb/src/main/java/org/apache/samza/storage/kv/RocksDbOptionsHelper.java [hunk 4]
```diff
@@ -42,59 +49,65 @@ public static Options options(Config storeConfig, SamzaContainerContext containe
...
-    int blockSize = storeConfig.getInt("rocksdb.block.size.bytes", 4096);
+
+    int blockSize = storeConfig.getInt(ROCKSDB_BLOCK_SIZE_BYTES, 4096);
     BlockBasedTableConfig tableOptions = new BlockBasedTableConfig();
     tableOptions.setBlockCacheSize(cacheSizePerContainer).setBlockSize(blockSize);
     options.setTableFormatConfig(tableOptions);
 
     CompactionStyle compactionStyle = CompactionStyle.UNIVERSAL;
-    String compactionStyleInConfig = storeConfig.get("rocksdb.compaction.style", "universal");
+    String compactionStyleInConfig = storeConfig.get(ROCKSDB_COMPACTION_STYLE, "universal");
```
Comment (vjagadish) [inline samza-kv-rocksdb/src/main/java/org/apache/samza/storage/kv/RocksDbOptionsHelper.java]: thanks!

File: docs/learn/documentation/versioned/jobs/configuration-table.html [hunk 5]
```diff
@@ -1422,6 +1422,21 @@
...
+                    <td class="default">2</td>
+                    <td class="description">
+                        The number of RocksDB LOG files (including rotated LOG.old.* files) to keep.
```
Comment (vjagadish) [inline docs/learn/documentation/versioned/jobs/configuration-table.html]: +1 for capitalized `LOG` (that way we know it's not the transaction `logfile`)

File: samza-kv-rocksdb/src/main/scala/org/apache/samza/storage/kv/RocksDbKeyValueStore.scala (hunk 1 above)
Comment (prateekm) [inline samza-kv-rocksdb/src/main/scala/org/apache/samza/storage/kv/RocksDbKeyValueStore.scala]: Yeah, remove and close are deprecated in newer versions.

File: gradle/dependency-versions.gradle (hunk 2 above)
Comment (prateekm) [inline gradle/dependency-versions.gradle]: My bad, meant SAMZA-984. Fixed.

File: docs/learn/documentation/versioned/jobs/configuration-table.html (hunk 3 above)
Comment (prateekm) [inline docs/learn/documentation/versioned/jobs/configuration-table.html]: Yeah, better to mirror RocksDB configuration as-is for these. I can rename the html id for consistency if you'd like.

Comment (vjagadish) [inline docs/learn/documentation/versioned/jobs/configuration-table.html]: Ah, no worries. Thanks for explaining.
